incubator/hnc/test/e2e/issues_test.go

```diff
@@ -237,3 +238,80 @@ var _ = Describe("Issues", func() {
 		mustRun("kubectl get rolebinding foo -n", nsChild, "-oyaml")
 	})
 })
+
+var _ = Describe("Issues that require reparing HNC", func() {
+
+	hncRecoverPath := os.Getenv("HNC_REPAIR")
+	const (
+		nsParent = "parent"
+		nsParent2 = "parent-2"
+		nsChild = "child"
+	)
+
+	BeforeEach(func() {// we don't want to destroy the HNC without being able to repair it, so skip this test if recovery path not set
+		if hncRecoverPath == ""{
+			Skip("Environment variable HNC_REPAIR not set. Skipping reocovering HNC.")
+		}
+		cleanupNamespaces(nsParent, nsParent2, nsChild)
+	})
+
+	AfterEach(func() {
+		cleanupNamespaces(nsParent, nsParent2, nsChild)
+		err := tryRun("kubectl apply -f", hncRecoverPath)
+		if err != nil {
+			GinkgoT().Log("-----------------------------WARNING------------------------------")
+			GinkgoT().Logf("WARNING: COULDN'T REPAIR HNC: %v", err)
+			GinkgoT().Log("ANY TEST AFTER THIS COULD FAIL BECAUSE WE COULDN'T REPAIR HNC HERE")
+			GinkgoT().Log("------------------------------------------------------------------")
+			GinkgoT().FailNow()
+		}
+		// give HNC enough time to repair
+		time.Sleep(5 * time.Second)
+	})
+
+	It("Should not delete child namespace when deleting a bad parent namespace - issue #797", func(){
```

Comment:
Can we be more specific than "bad parent namespace"? I _think_ this is "namespace with bad anchor"?

Reply:
done